numpy/lib/financial.py

```diff
@@ -208,12 +215,15 @@ def pmt(rate, nper, pv, fv=0, when='end'):
     """
     when = _convert_when(when)
     (rate, nper, pv, fv, when) = map(np.asarray, [rate, nper, pv, fv, when])
-    temp = (1 + rate)**nper
-    mask = (rate == 0.0)
-    np.copyto(rate, 1.0, where=mask)
-    z = np.zeros(np.broadcast(rate, nper, pv, fv, when).shape)
-    fact = np.where(mask != z, nper + z, (1 + rate*when)*(temp - 1)/rate + z)
-    return -(fv + pv*temp) / fact
+
+    # use different payment if rate == 0
+    zero_rate_mask = rate == 0
```

Comment:
Ok updated the code per your suggestion and ran "git pull --force"

On Mon, Jul 13, 2015 at 3:53 PM, Nathaniel J. Smith <
notifications@github.com> wrote:

> In numpy/lib/financial.py
> <https://github.com/numpy/numpy/pull/6073#discussion_r34510377>:
>
> > @@ -208,12 +215,15 @@ def pmt(rate, nper, pv, fv=0, when='end'):
> >      """
> >      when = _convert_when(when)
> >      (rate, nper, pv, fv, when) = map(np.asarray, [rate, nper, pv, fv, when])
> > -    temp = (1 + rate)**nper
> > -    mask = (rate == 0.0)
> > -    np.copyto(rate, 1.0, where=mask)
> > -    z = np.zeros(np.broadcast(rate, nper, pv, fv, when).shape)
> > -    fact = np.where(mask != z, nper + z, (1 + rate*when)*(temp - 1)/rate + z)
> > -    return -(fv + pv*temp) / fact
> > +
> > +    # use different payment if rate == 0
> > +    zero_rate_mask = rate == 0
>
> Please put parens around the == expression to make it easier to read: zero_rate_mask
> = (rate == 0)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/6073/files#r34510377>.
>